### PR TITLE
Implement AD Trust checking

### PR DESCRIPTION
**Pull request by rcritten**
This checks a bunch of ADTrust options as outlined by Alexander.

The DS change is there because I set the wrong container originally and changing it to be right for AD Trust requires it be changed here as well. It's pedantic but could be important at some point.

### CODE REVIEW

**Comment by rcritten** [comment]
@fcami note that I'm toying with a new system for the response in the output to use a formatted string. If this looks ok we can apply it to the others as well (this will be important for localization eventually). It also forced me to be sure that all variables were listed in the keywords.

File: src/ipahealthcheck/ipa/trust.py [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+                             msg='{key} service is not enabled')
+
+
+@registry
+class IPATrustControllerConfCheck(IPAPlugin):
+    """
+    Verify that smb.conf matches the template
+    """
+    @duration
+    def check(self):
```
Comment (abbra) [inline src/ipahealthcheck/ipa/trust.py]: It is just fine to have more options in the `net conf list` than what we have in the `smb.conf.template`.  Also, some options aren't really important to see exact. For example, `log level` might be any. I'm not sure how to achieve this...

File: src/ipahealthcheck/ipa/trust.py [hunk 2]
```diff
@@ -0,0 +1,499 @@
...
+                             capture_output=True)
+        if result.returncode != 0:
+            yield Result(self, constants.ERROR,
+                         key='domain_list_error',
+                         sslctl=paths.SSSCTL,
+                         error=result.error_log,
+                         msg='Execution of {sslctl} failed: {error}')
+            return
+        sssd_domains = result.output.strip().split('\n')
+        if 'implicit_files' not in sssd_domains:
```
[the remark this answers is not in the document]
Comment (rcritten) [reply src/ipahealthcheck/ipa/trust.py]: Ok I can drop that check, though it would always pass :-)

File: src/ipahealthcheck/ipa/trust.py (hunk 1 above)
Comment (rcritten) [reply src/ipahealthcheck/ipa/trust.py]: Should we punt on this then? Perhaps we can come up with a list of absolutely-required settings that I can validate.

**Comment by rcritten** [comment]
The suggested changes applied plus a new mechanism to identify whether the current host is a trust agent or controller using servroles directly.

**Comment by abbra** [comment]
LGTM. ACK.

File: src/ipahealthcheck/ipa/trust.py (hunk 1 above)
Comment (abbra) [reply src/ipahealthcheck/ipa/trust.py]: well, there are few key ones. We can do them in a separate pull request. I'll build a list.

**Comment by fcami** [comment]
LGTM too. ACK.

**Pull request merged**